WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2167,7 +2173,26 @@ static GSourceFuncs _handlerIntervention =
                 }
                 g_mkdir_with_parents(wpeDiskCachePath, 0700);
 
+#ifdef WEBKIT_MEMORY_PRESSURE_API
+                if (_config.NetworkProcessLimit.IsSet() == true) {
+                    WebKitMemoryPressureSettings* memoryPressureSettings = webkit_memory_pressure_settings_new();
+                    webkit_memory_pressure_settings_set_memory_limit(memoryPressureSettings, _config.NetworkProcessLimit.Value());
+                    webkit_website_data_manager_set_memory_pressure_settings(memoryPressureSettings);
+                    webkit_memory_pressure_settings_free(memoryPressureSettings);
+                }
+
+                WebKitWebsiteDataManager* websiteDataManager;
+                if (_config.WebProcessLimit.IsSet() == true) {
+                    WebKitMemoryPressureSettings* memoryPressureSettings = webkit_memory_pressure_settings_new();
+                    webkit_memory_pressure_settings_set_memory_limit(memoryPressureSettings, _config.WebProcessLimit.Value());
+                    websiteDataManager = webkit_website_data_manager_new("local-storage-directory", wpeStoragePath, "disk-cache-directory", wpeDiskCachePath, "memory-pressure-settings", memoryPressureSettings, nullptr);
```

Comment:
Ozgur will check if we only have the webkit_website_data_manager_new api. Perhaps we can split the directory settings from the memory settings and make the code a little more clear here

Reply:
Discussed with Ozgur.